### PR TITLE
Source Hubspot: handle 400/403 errors with logger warnings

### PR DESCRIPTION
**Pull request by arsenlosenko**
## What
Resovle P0 issue that was raised due to previous changes:
https://github.com/airbytehq/oncall/issues/2677

## How
Previous changes in error handling that I made were relying on raising of `config_error` exceptions  when we receive 400/403 errors from Hubspot API. Before that, we didn't handle 400 error and just passed error message from Hubspot API directly. By introducing my changes with `config_error` it changed the UX and in case of such errors user's syncs would stop execution. In order to avoid that, I rewrote handling of 400/403 errors to just logger warnings. This way user will receive a message that some of his permissions or configuration does not allow to sync certain streams, but will keep syncing other available streams.

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`0a6ddc91b6`](https://github.com/airbytehq/airbyte/pull/29211/commits/0a6ddc91b6802c69cf98f244f773e864c322aec6)) - ❌

⏲️  Total pipeline duration: 13mn00s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-hubspot-fix-p0-403-error-handling/1691506216/0a6ddc91b6802c69cf98f244f773e864c322aec6/source-hubspot/1.3.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:0a6ddc91b6802c69cf98f244f773e864c322aec6)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`892a17fb29`](https://github.com/airbytehq/airbyte/pull/29211/commits/892a17fb29f7379901765158428487a557db8e82)) - ✅

⏲️  Total pipeline duration: 12mn19s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-hubspot-fix-p0-403-error-handling/1691507996/892a17fb29f7379901765158428487a557db8e82/source-hubspot/1.3.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:892a17fb29f7379901765158428487a557db8e82)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 1]
```diff
@@ -193,22 +185,23 @@ def _parse_and_handle_errors(response) -> Union[MutableMapping[str, Any], List[M
...
-            message = f"The authenticated user does not have permissions to access the URL {response.url}. Verify your permissions to access this endpoint."
-            raise HubspotAccessDenied(internal_message=message, failure_type=FailureType.config_error, response=response)
+            logger.warning(
+                f"The authenticated user does not have permissions to access the URL {response.url}. Verify your permissions to access this endpoint."
+            )
         elif response.status_code in (HTTPStatus.UNAUTHORIZED, CLOUDFLARE_ORIGIN_DNS_ERROR):
             message = (
                 "The user cannot be authorized with provided credentials. Please verify that your credentails are valid and try again."
             )
-            raise HubspotInvalidAuth(internal_message=message, failure_type=FailureType.config_error, response=response)
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:205]: Why you delete failure_type config error?

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 2]
```diff
@@ -193,22 +185,23 @@ def _parse_and_handle_errors(response) -> Union[MutableMapping[str, Any], List[M
             message = response.json().get("message")
 
         if response.status_code == HTTPStatus.BAD_REQUEST:
-            message = f"Request to {response.url} didn't succeed. Please verify your credentials and try again.\nError message from Hubspot API: {message}"
```
[the remark this answers is not in the document]
Comment (arsenlosenko) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:196]: removed it on mistake, reverted it back

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 3]
```diff
@@ -193,22 +185,23 @@ def _parse_and_handle_errors(response) -> Union[MutableMapping[str, Any], List[M
...
             message = (
                 "The user cannot be authorized with provided credentials. Please verify that your credentails are valid and try again."
             )
-            raise HubspotInvalidAuth(internal_message=message, failure_type=FailureType.config_error, response=response)
+            raise HubspotInvalidAuth(message, response=response)
         elif response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
             retry_after = response.headers.get("Retry-After")
             raise HubspotRateLimited(
-                internal_message=f"You have reached your Hubspot API limit. We will resume replication once after {retry_after} seconds."
+                f"You have reached your Hubspot API limit. We will resume replication once after {retry_after} seconds."
```
[the remark this answers is not in the document]
Comment (arsenlosenko) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py]: done

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py (hunk 1 above)
Comment (arsenlosenko) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:205]: removed it by mistake, added back

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`f38d27e571`](https://github.com/airbytehq/airbyte/pull/29211/commits/f38d27e57179c4488870496b96e02f8fbdecbdfd)) - ❌

⏲️  Total pipeline duration: 11mn05s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ❌        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-hubspot-fix-p0-403-error-handling/1691666499/f38d27e57179c4488870496b96e02f8fbdecbdfd/source-hubspot/1.3.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f38d27e57179c4488870496b96e02f8fbdecbdfd)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`130ae59080`](https://github.com/airbytehq/airbyte/pull/29211/commits/130ae59080068f330917faa87312e5fed145c5c3)) - ✅

⏲️  Total pipeline duration: 11mn42s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-hubspot-fix-p0-403-error-handling/1691667368/130ae59080068f330917faa87312e5fed145c5c3/source-hubspot/1.3.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:130ae59080068f330917faa87312e5fed145c5c3)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Review by lazebnyi — APPROVED** [review]

**Review by clnoll — APPROVED** [review]
LGTM @arsenlosenko!

**Comment by octavia-squidington-iii** [comment]
##  source-hubspot test report (commit [`5ef0485f34`](https://github.com/airbytehq/airbyte/pull/29211/commits/5ef0485f348062651c240feb1a34a06f93bce0c5)) - ✅

⏲️  Total pipeline duration: 12mn35s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-hubspot/metadata.yaml | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |
| Code format checks                                                    | ✅        |
| Connector package install                                             | ✅        |
| Build source-hubspot docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                            | ✅        |
| Integration tests                                                     | ✅        |
| Acceptance tests                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/arsenlosenko_source-hubspot-fix-p0-403-error-handling/1691684441/5ef0485f348062651c240feb1a34a06f93bce0c5/source-hubspot/1.3.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:5ef0485f348062651c240feb1a34a06f93bce0c5)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-hubspot test
```

**Pull request merged**